### PR TITLE
[chore] [processor/tailsampling] Clarify ordering relative to context-dependent processors (#23870)

### PR DESCRIPTION
**Pull request by gebn**
**Description:** Makes clear that `tail_sampling` must be placed after `k8sattributes` and other processors that clobber context. Please see issue below for why this is an easy trap to fall into.

**Link to tracking Issue:** #23870

### CODE REVIEW

**Review by jpkrohling — APPROVED** [review]

**Comment by jpkrohling** [comment]
@gebn, looks like you still need to sign the CLA

**Comment by gebn** [comment]
Thanks @jpkrohling, just waiting for corporate sign-off on the CLA.

File: .chloggen/k8sattrs-tailsampling-docs.yaml
Comment (dmitryax) [inline .chloggen/k8sattrs-tailsampling-docs.yaml:20]: We don't need changelog entries for the doc updates. Please remove it. `[chore]` tag in the PR header suppresses the CI check

Comment (gebn) [reply .chloggen/k8sattrs-tailsampling-docs.yaml]: Removed, and also rebased to latest `main`.

**Pull request merged**
